tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx

```diff
@@ -34,6 +39,18 @@ public:
       REntryRange(Long64_t startEntry, Long64_t endEntry);
    };
 
+   // groups need to fulfill:
+   // 1. preserve the original order -> arrange them in a vector, store also number of fileglobs in this group
+   // 2. there is 1:1 correspondence between group and meta data => group and metadata can go together
+   // Current solution: create a simple struct to hold the triplet {groupName, groupSize, MetaData}
+   // This Group structure does not need an exposure to the user.
+   struct RGroupInfo {
```

Comment:
we have `RGroupMetaData`, `RGroupInfo` and `RSampleSet` with a large amount of overlap.

can we merge them in a single `RDatasetGroup` (or whatever name we pick) that stores, for each group:

- unique group id
- group name
- tree names
- file names
- metadata

?

Reply:
Yes, this is what I just did in the latest commit!